src/Microsoft.Health.SqlServer/Features/Client/SqlServerTransientFaultRetryPolicyFactory.cs

```diff
@@ -37,8 +37,9 @@ internal class SqlServerTransientFaultRetryPolicyFactory : ISqlServerTransientFa
                 transientFaultRetryPolicyConfiguration.Factor,
                 transientFaultRetryPolicyConfiguration.FastFirst);
 
+            // Check for exceptions that are considered transient by SQL Server, deadlocks (1205), or timeouts.
             PolicyBuilder policyBuilder = Policy
-                .Handle<SqlException>(sqlException => sqlException.IsTransient())
+                .Handle<SqlException>(sqlException => sqlException.IsTransient() || sqlException.Number == 1205)
```

Comment:
Are deadlocks transient in general? I would think we would need some mechanism for resolving the deadlock.

Reply:
Yes, we should definitely keep an eye out for deadlocks and do what we can to resolve them. This should alleviate the error the end user will currently see if they encounter an operation with a deadlock. Below is an example of what is logged when a deadlock is encounter after this change, so it should still show up in our logs.

```
warn: PollyRetryLogger[0]
      The operation failed. Will retry in '00:00:00'. Retried 1 of time(s) so far.
      Microsoft.Data.SqlClient.SqlException (0x80131904): Transaction (Process ID 77) was deadlocked on lock resources with another process and has been chosen as the deadlock victim. Rerun the transaction.
         at Microsoft.Data.SqlClient.SqlConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)
         at Microsoft.Data.SqlClient.SqlInternalConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)
         at Microsoft.Data.SqlClient.TdsParser.ThrowExceptionAndWarning(TdsParserStateObject stateObj, Boolean callerHasConnectionLock, Boolean asyncClose)
         at Microsoft.Data.SqlClient.TdsParser.TryRun(RunBehavior runBehavior, SqlCommand cmdHandler, SqlDataReader dataStream, BulkCopySimpleResultSet bulkCopyHandler, TdsParserStateObject stateObj, Boolean& dataReady)
         at Microsoft.Data.SqlClient.SqlDataReader.TryConsumeMetaData()
         at Microsoft.Data.SqlClient.SqlDataReader.get_MetaData()
         at Microsoft.Data.SqlClient.SqlCommand.FinishExecuteReader(SqlDataReader ds, RunBehavior runBehavior, String resetOptionsString, Boolean isInternal, Boolean forDescribeParameterEncryption, Boolean shouldCacheForAlwaysEncrypted)
         at Microsoft.Data.SqlClient.SqlCommand.CompleteAsyncExecuteReader(Boolean isInternal, Boolean forDescribeParameterEncryption)
         at Microsoft.Data.SqlClient.SqlCommand.InternalEndExecuteReader(IAsyncResult asyncResult, Boolean isInternal, String endMethod)
         at Microsoft.Data.SqlClient.SqlCommand.EndExecuteReaderInternal(IAsyncResult asyncResult)
         at Microsoft.Data.SqlClient.SqlCommand.EndExecuteReaderAsync(IAsyncResult asyncResult)
         at Microsoft.Data.SqlClient.SqlCommand.EndExecuteReaderAsyncCallback(IAsyncResult asyncResult)
         at System.Threading.Tasks.TaskFactory`1.FromAsyncCoreLogic(IAsyncResult iar, Func`2 endFunction, Action`1 endAction, Task`1 promise, Boolean requiresSynchronization)
      --- End of stack trace from previous location ---
         at Polly.Retry.AsyncRetryEngine.ImplementationAsync[TResult](Func`3 action, Context context, CancellationToken cancellationToken, ExceptionPredicates shouldRetryExceptionPredicates, ResultPredicates`1 shouldRetryResultPredicates, Func`5 onRetryAsync, Int32 permittedRetryCount, IEnumerable`1 sleepDurationsEnumerable, Func`4 sleepDurationP
rovider, Boolean continueOnCapturedContext)
      ClientConnectionId:df28203e-20ed-4e65-afb4-3718e7610399
      Error Number:1205,State:51,Class:13
```

Reply:
Ah! I see. Certainly the `Rerun the transaction.` is what we want to do too lol

Reply:
Should `sqlException.Number == 1205` be included in `sqlException.IsTransient()` then :)?

Reply:
For whatever reason, I totally did not connect that `IsTransient()` was our own extension method. Yes, it makes sense there.